File: backend/create-data.py
import random
import csv

headers = ["Kid Age","Kid Gender","Kid Ethnicity","Kid Location", "Kid Hobbies", "Kid Health Conditions", "Kid Number of Siblings", "Kid Education", "Adult Age", "Adult Gender", "Adult Ethnicity", "Adult Location", "Adult Marital Status", "Adult Income", "Adult Occupation", "Adult Health Conditions", "Compatibility"]


  
num_of_samples = 1000


ethnicities = ['African', 'African American', 'Arab', 'Asian', 'Caribbean', 'European', 'Hispanic/Latino', 'Indigenous', 'Middle Eastern', 'Native American', 'Pacific Islander', 'South Asian', 'Southeast Asian'];

data =[]
data.append(headers)
for _ in range(num_of_samples):
    data.append([random.randint(1, 18), random.randint(0,1), random.randint(1, len(ethnicities)), random.randint(501, 99950), random.randint(1, 10), random.randint(1,10), random.randint(0,5), random.randint(1,4), random.randint(18, 90), random.randint(0,1), random.randint(1, len(ethnicities)), random.randint(501, 99950), random.randint(0,1), random.randint(25750, 154500), random.randint(1,20), random.randint(1,10), random.randint(0,1)])

with open("data.csv", mode='w', newline='') as file:
    writer = csv.writer(file)
    writer.writerows(data)

File: backend/create-data.js
let headers = ["Kid Age","Kid Gender","Kid Ethnicity","Kid Location", "Kid Hobbies", "Kid Health Conditions", "Kid Number of Siblings", "Kid Education", "Adult Age", "Adult Gender", "Adult Ethnicity", "Adult Location", "Adult Marital Status", "Adult Income", "Adult Occupation", "Adult Health Conditions", "Compatibility"].join("\t");

function randi(min, max) { // min and max included 
    return Math.floor(Math.random() * (max - min + 1) + min);
}
  
let num_of_samples = 1000;

let content = headers + "\n";

//TODO change
let ethnicities = ['African', 'African American', 'Arab', 'Asian', 'Caribbean', 'European', 'Hispanic/Latino', 'Indigenous', 'Middle Eastern', 'Native American', 'Pacific Islander', 'South Asian', 'Southeast Asian'];



for (let i = 0; i < num_of_samples; i++) {
    content += [randi(1, 18), randi(0,1), randi(1, ethnicities.length), randi(501, 99950), randi(1, 10), randi(1,10), randi(0,5), randi(1,4), randi(18, 90), randi(0,1), randi(1, ethnicities.length), randi(501, 99950), randi(0,1), randi(25750, 154500), randi(1,20), randi(1,10), randi(0,1)].join('\t') + '\n';
}

let fs = require("fs");

fs.writeFile("data.csv", content, (err) => {
    if (err) {
        console.error(err);
        return;
    }
    console.log("File written successfully");
});